src/benchmarklib/tpch/tpch_table_generator.cpp

```diff
@@ -220,10 +221,14 @@ TpchTableGenerator::TpchTableGenerator(float scale_factor, const std::shared_ptr
 std::unordered_map<std::string, BenchmarkTableInfo> TpchTableGenerator::generate() {
   Assert(!_benchmark_config->cache_binary_tables, "Caching binary Tables not supported by TpchTableGenerator, yet");
 
-  const auto customer_count = static_cast<size_t>(tdefs[CUST].base * _scale_factor);
-  const auto order_count = static_cast<size_t>(tdefs[ORDER].base * _scale_factor);
-  const auto part_count = static_cast<size_t>(tdefs[PART].base * _scale_factor);
-  const auto supplier_count = static_cast<size_t>(tdefs[SUPP].base * _scale_factor);
+  // Init tpch_dbgen - it is important this is done before any data structures from tpch_dbgen are read.
+  dbgen_reset_seeds();
+  dbgen_init_scale_factor(_scale_factor);
```

Comment:
Do I understand it correctly that we undo the `scale` changes and use the functions closer to the way dbgen intended it?

Reply:
yes